demo/java/src/main/java/io/github/cosinekitty/astronomy/demo/LunarEclipse.java

```diff
@@ -11,15 +13,11 @@ public class LunarEclipse {
      */
     public static int run(Time startTime) {
         LunarEclipseInfo e = Astronomy.searchLunarEclipse(startTime);
```

Comment:
I'm starting to think I should also provide generic iterators for everything where I have a (searchX, nextX) pair of functions. I don't know about the naming, but something like

```java
Astronomy.lunarEclipsesAfter(startTime)
    .filter(x -> x.getKind() != EclipseKind.Penumbral)
    .limit(10)
    .forEach(LunarEclipse::printEclipse);
```

This would be very easy to do in C# and Python, and it looks like it should be easy to do in Java and Kotlin also. I will keep the existing search/next functions, because the work is already done and they could be useful. What do you think?

Reply:
I like this kind of API!